UploadDaemon_Test/SymbolAnalysis/SymbolCollectionResolverTest.cs

```diff
@@ -0,0 +1,80 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UploadDaemon.Configuration;
+
+namespace UploadDaemon.SymbolAnalysis
+{
+    [TestFixture]
+    class SymbolCollectionResolverTest
+    {
+        private readonly GlobPatternList someAssemblyPatterns = new GlobPatternList(new List<string> { "*" }, new List<string> { });
+
+        private SymbolCollectionResolver resolver;
+
+        [SetUp]
+        public void SetUp()
+        {
+            resolver = new SymbolCollectionResolver();
+        }
+
+        [Test]
+        public void CollectsSymbolFileNames()
+        {
+            SymbolCollection collection = resolver.ResolveFrom(TestUtils.TestDataDirectory, someAssemblyPatterns);
+
+            Assert.That(Normalized(collection.SymbolFileNames), Is.EqualTo(new[] { "\\test-data\\UploadDaemon.SymbolAnalysis.SymbolCollectionResolverTest\\Some.pdb" }));
+        }
+
+        [Test]
+        public void CachesCollection()
+        {
+            SymbolCollection collection1 = resolver.ResolveFrom(TestUtils.TestDataDirectory, someAssemblyPatterns);
+            SymbolCollection collection2 = resolver.ResolveFrom(TestUtils.TestDataDirectory, someAssemblyPatterns);
+
+            Assert.That(collection1, Is.SameAs(collection2));
+        }
+
+        [Test]
+        public void InvalidatesCacheIfSymbolFileChanges()
+        {
+            SymbolCollection collection1 = resolver.ResolveFrom(TestUtils.TestDataDirectory, someAssemblyPatterns);
+
+            SimulateSymbolFileChange(collection1.SymbolFileNames.First());
+
+            SymbolCollection collection2 = resolver.ResolveFrom(TestUtils.TestDataDirectory, someAssemblyPatterns);
+
+            Assert.That(collection1, Is.Not.SameAs(collection2));
+        }
+
+        [Test]
+        public void CachesCollectionAgainAfterInvalidation()
```

Comment:
please add a test for a file deletion

Reply:
Added tests for new PDBs and deleting PDBs. Deletion actually worked already. New files now work as well.